machinist/mserver/controller.go

```diff
@@ -143,27 +143,36 @@ func (en *Controller) addNodeToDns(name string, ips []net.IP, tags []string) {
 	}
 }
 
-// ServeAllRecords will continuously poll Nodes() and create multiple _all.<domain> records containing the ip addresses
+// ServeAllAndInfoRecords will continuously poll Nodes() and create multiple _all.<domain> records containing the ip addresses
 // of all machines attached.
+// It also serves the current state of the dns server via the _info record.
 // TODO(adam): be able to pass in a wrapped ticker for testing intervals
-func (en *Controller) ServeAllRecords(killChannel chan struct{}, killChannelAck chan struct{}) {
+func (en *Controller) ServeAllAndInfoRecords(killChannel chan struct{}, killChannelAck chan struct{}) {
 	for {
 		select {
 		case <-time.After(en.allRecordsRefreshRate):
 			ns := en.Nodes()
 			for _, d := range en.dnsServer.Domains {
 				dnsName := dns.CanonicalName(fmt.Sprintf("%s.%s", "_all", d))
-				var rs []dns.RR
+				infoDnsName := dns.CanonicalName(fmt.Sprintf("%s.%s", "_info", d))
```

Comment:
Seems like this section could be streamlined using another for loop that iterates over []string{"all", "info"}. The only thing that varies based on the current iterator value is the string format used for the call to dns.NewRR().